vaadin-combo-box-flow-integration-tests/src/test/java/com/vaadin/flow/component/combobox/test/ClearValueIT.java

```diff
@@ -41,6 +40,23 @@ public void valueIsCorrectlyCleared() {
                 ClearValuePage.BUTTON_CLEAR_ID, false);
     }
 
+    @Test
+    public void valueIsCorrectlyClearedWithClearButtonBeforeOpened() {
+        String comboBoxId = ClearValuePage.COMBO_BOX_WITH_CLEAR_BUTTON_ID;
+        ComboBoxElement comboBox = $(ComboBoxElement.class).id(comboBoxId);
+        Assert.assertEquals(String.format(
+                "Unexpected selected item label for combo box with id '%s'",
+                comboBoxId), ClearValuePage.INITIAL_VALUE,
+                comboBox.getSelectedText());
+
+        comboBox.$("vaadin-text-field").get(0).$("[part~='clear-button']").get(0).click();
+        comboBox.openPopup();
+
+        Assert.assertEquals(String.format(
+                "Combo box with id '%s' should have its value empty after the test",
+                comboBoxId), "", comboBox.getSelectedText());
```

Comment:
This verifies only that the text-field is empty in client-side. It should verify the server-side state of `ComboBox.getValue()`. You need to add a value-change-listener or a button which sets some text on the page to e.g. "null" if the combo box's value is `null`

Reply:
Done